### PR TITLE
conda-smithy 3.35.0

### PR DESCRIPTION
**Pull request by beckermr**
Merge only after success.

This pull request was auto-generated by [rever](https://regro.github.io/rever-docs/)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-{% set version = "3.34.1" %}
+{% set version = "3.35.0" %}
 
 package:
   name: conda-smithy
   version: {{ version }}
 
 source:
   url: https://github.com/conda-forge/conda-smithy/releases/download/v{{ version }}/conda-smithy-{{ version }}.tar.gz
```
Comment (beckermr) [inline recipe/meta.yaml:8]:
```suggestion
  url: https://github.com/conda-forge/conda-smithy/releases/download/v{{ version }}/conda_smithy-{{ version }}.tar.gz
```

**Comment by h-vetinari** [comment]
Looking at https://github.com/conda-forge/conda-smithy/blame/main/environment.yml, we need something like the following here:
```diff
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -30,13 +30,14 @@ requirements:
     - python >=3.6
     - setuptools
     - conda >=4.2
-    - conda-build >=3.18.3
+    - conda-build >=24.3
     - conda-package-handling >=1.9.0
     - jinja2
     - requests
     - pycryptodome
     - gitpython
     - git
+    - pydantic >=2,<3
     - pygithub >=2,<3
     - ruamel.yaml >=0.16
     - conda-forge-pinning
@@ -46,7 +47,9 @@ requirements:
     - license-expression
     - libarchive
     - tomli >=1.0.0  # for python <3.11, can remove in the future
+    - backports.strenum
     - cirun >=0.30
+    - exceptiongroup
     - jsonschema
   run_constrained:
     # For more details about `shellcheck`, please see this issue.
```

We also might want to sort the runtime dependencies alphabetically...

**Comment by beckermr** [comment]
I am pretty sure pydantic is not a runtime dep. @isuruf has been keeping track of that and should be able to say.

**Comment by beckermr** [comment]
exception group appears to be optional as well?

the relevant code section is

```
try:
    from builtins import ExceptionGroup
except ImportError:
    from exceptiongroup import ExceptionGroup
```

**Comment by h-vetinari** [comment]
> exception group appears to be optional as well?

No that just means it'll use the 3.11 builtin if available, and fall back to the backport otherwise (which needs to be there, or we fail also in the `except:` branch).

**Review by h-vetinari — APPROVED** [review]

**Pull request merged**